### PR TITLE
Stock/delete stockitem cascading to horses

### PR DESCRIPTION
**Pull request by ChristianBorn**

### CODE REVIEW

File: backend/src/test/java/de/ffmjava/capstone/backend/stock/StockServiceTest.java [hunk 1]
```diff
@@ -73,35 +79,63 @@ void addNewStockItem_AndExpectStockItem_200() {
...
         }
     }
 
+    @Test
+    void deleteStockItem_noConsumptionToDelete_200() {
+        String idToDelete = "1";
+        when(mockStockRepository.existsById(idToDelete))
+                .thenReturn(true);
+        when(mockHorseRepository.findHorsesByConsumptionId("1")).thenReturn(new ArrayList<>());
+        assertTrue(service.deleteStockItem("1"));
```
Comment (ToLa09) [inline backend/src/test/java/de/ffmjava/capstone/backend/stock/StockServiceTest.java:119]: without given/when/then its harder to read. Overlooked multiple times that in line 119 the service is called

File: backend/src/test/java/de/ffmjava/capstone/backend/stock/StockServiceTest.java [hunk 2]
```diff
@@ -73,35 +79,63 @@ void addNewStockItem_AndExpectStockItem_200() {
...
+
+        when(mockStockRepository.existsById(idToDelete))
+                .thenReturn(true);
+        when(mockHorseRepository.findHorsesByConsumptionId(idToDelete))
+                .thenReturn(retrievedHorses);
+        when(mockHorseRepository.saveAll(List.of(horseWithoutConsumption)))
+                .thenReturn(List.of(horseWithoutConsumption));
+
+        assertTrue(service.deleteStockItem("1"));
+        verify(mockHorseRepository).findHorsesByConsumptionId(any());
```
Comment (ToLa09) [inline backend/src/test/java/de/ffmjava/capstone/backend/stock/StockServiceTest.java:138]: also without given/when/then harder to read. Again overlooked where the service gets called

File: frontend/src/horses/HorseOverview.tsx [hunk 3]
```diff
@@ -62,8 +62,9 @@ function HorseOverview() {
     }
 
     const removeConsumption = (consumptionItemToDeleteId: string, editedHorse: HorseModel) => {
-        editedHorse.consumptionList = editedHorse.consumptionList
-            .filter(consumptionItem => consumptionItem.id !== consumptionItemToDeleteId)
+        console.log(editedHorse.consumptionList)
+        editedHorse.consumptionList = editedHorse.consumptionList.filter(consumptionItem => consumptionItem.id !== consumptionItemToDeleteId)
+        console.log(editedHorse.consumptionList)
```
Comment (ToLa09) [inline frontend/src/horses/HorseOverview.tsx:67]: console.log() doesn't belong to production code :)

File: backend/src/main/java/de/ffmjava/capstone/backend/stock/StockRepository.java [hunk 4]
```diff
@@ -2,7 +2,9 @@
 
 import de.ffmjava.capstone.backend.stock.model.StockItem;
 import org.springframework.data.mongodb.repository.MongoRepository;
+import org.springframework.stereotype.Repository;
 
+@Repository
 public interface StockRepository extends MongoRepository<StockItem, String> {
     StockItem getById(String id);
```
[the remark this answers is not in the document]
Comment (ChristianBorn) [reply backend/src/main/java/de/ffmjava/capstone/backend/stock/StockRepository.java:9]: Good point

**Review by IrynaKamenska — APPROVED** [review]

**Pull request merged**